dev_guide/expanding_persistent_volumes.adoc

```diff
@@ -0,0 +1,79 @@
+** If the volume is of a type that requires a file system resize, then a file system resize is performed on kubelet. 
+* If the volume plugin is of type that cannot do resizing of attached volumes (such as Cinder), then `ExpandVolumeDevice` can return an error by checking for volume status with its own API (such as by making an Openstack Cinder API call in this case). The controller continues trying to resize the volume until it is successful.
+* To consider cases of missed PVC update events, an additional loop reconciles bound PVCs with PVs. This additional loop loops through all PVCs, matches `pvc.spec.resources.requests` with `pv.spec.capacity`, and adds the PVC in the volume_expand_controller's work queue if `pv.spec.capacity` is less than `pvc.spec.resources.requests`.
+* There are additional checks in the controller that expand the PV size. These checks ensure that you do not make volume plugin API calls that can reduce the size of the PV.
+
+[[expanding-pvcs]]
+== Expanding Persistent Volumes Claims
+
+Kubernetes 1.8 added Alpha support for expanding persistent volumes. In Kubernetes 1.9, the following volume types support expanding PVCs:
+
+* gcePersistentDisk
+* awsElasticBlockStore
+* Cinder
+* glusterfs
+* rbd
+
+You can allow expanding persistent volume claims by setting the `ExpandPersistentVolumes` feature gate to true. You should also enable the `PersistentVolumeClaimResize` admission plugin to perform additional validations of volumes that can be resized.
+
+Once the `PersistentVolumeClaimResize` admission plug-in has been turned on, resizing is only allowed for storage classes whose `allowVolumeExpansion` field is set to true, as shown below.
+
+----
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: gluster-vol-default
+provisioner: kubernetes.io/glusterfs
+parameters:
+  resturl: "http://192.168.10.100:8080"
+  restuser: ""
+  secretNamespace: ""
+  secretName: ""
+allowVolumeExpansion: true
+----
+
+Once both the feature gate and the aforementioned admission plug-in are turned on, users can request a larger volume for their `PersistentVolumeClaim` by editing the claim and requesting a larger size. This triggers expansion of the volume that is backing the underlying `PersistentVolume`. A new `PersistentVolume` is not created to satisfy the claim. Kubernetes instead attempts to resize the existing volume.
+
+To expand volumes containing a file system, file system resizing is only performed when a new pod is started using the `PersistentVolumeClaim` in `ReadWrite` mode. If a volume being expanded is used in a pod or deployment, you need to delete and recreate the pod for file system resizing to take place. Also, file system resizing is only supported for XFS, Ext3, and Ext4 file system types.
```

Comment:
Lets add a new section for Expanding volume types with file system. Such as - EBS, GCE-PD, Cinder and RBD. That section should clearly document 2 step process.

Step 1: Edit your PVC to have new size. User should wait for PVC to have condition `PersistentVolumeClaimFileSystemResizePending` 
Step 2: User can describe the PVC and when the PVC has condition `PersistentVolumeClaimFileSystemResizePending` then user can (re-)start a pod to finish file system resize of volume on node.